src/code/z_camera.c

```diff
@@ -543,323 +866,7424 @@ s32 Camera_GetWaterBoxBgCamSetting(Camera* camera, f32* waterY) {
+    return ret;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_Subject0.s")
+/**
+ * Calculates how much to adjust the camera at's y value when on a slope.
+ */
+f32 Camera_CalcSlopeYAdj(Vec3f* floorNorm, s16 playerYRot, s16 eyeAtYaw, f32 adjAmt) {
+    f32 tmp;
+    VecSph floorNormSph;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_Data0.s")
+    OLib_Vec3fToVecSphGeo(&floorNormSph, floorNorm);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_Data1.s")
+    tmp = Math_CosS(floorNormSph.pitch) * Math_CosS(playerYRot - floorNormSph.yaw);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_Data2.s")
+    return (Camera_fabsf(tmp) * adjAmt) * Math_CosS(playerYRot - eyeAtYaw);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_Data3.s")
+f32 func_800CD6CC(Actor* actor) {
+    return sqrtf(gTargetRanges[actor->targetMode].rangeSq / gTargetRanges[actor->targetMode].leashScale);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_Data4.s")
+/**
+ * Calculates new at vector for the camera pointing in `eyeAtDir`
+ */
+s32 Camera_CalcAtDefault(Camera* camera, VecSph* eyeAtDir, f32 yOffset, s16 calcSlope) {
+    Vec3f* at = &camera->at;
+    Vec3f focalActorAtOffsetTarget;
+    Vec3f atTarget;
+    s32 pad;
+    PosRot* focalActorPosRot = &camera->focalActorPosRot;
+    f32 focalActorHeight = Camera_GetFocalActorHeight(camera);
+
+    focalActorAtOffsetTarget.y = focalActorHeight + yOffset;
+    focalActorAtOffsetTarget.x = 0.0f;
+    focalActorAtOffsetTarget.z = 0.0f;
+
+    if (calcSlope) {
+        focalActorAtOffsetTarget.y -= OLib_ClampMaxDist(
+            Camera_CalcSlopeYAdj(&camera->floorNorm, focalActorPosRot->rot.y, eyeAtDir->yaw, 25.0f), focalActorHeight);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_Unique1.s")
+    Camera_ScaledStepToCeilVec3f(&focalActorAtOffsetTarget, &camera->focalActorAtOffset, camera->xzOffsetUpdateRate,
+                                 camera->yOffsetUpdateRate, 0.1f);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_Unique2.s")
+    atTarget.x = focalActorPosRot->pos.x + camera->focalActorAtOffset.x;
+    atTarget.y = focalActorPosRot->pos.y + camera->focalActorAtOffset.y;
+    atTarget.z = focalActorPosRot->pos.z + camera->focalActorAtOffset.z;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_Unique3.s")
+    if (atTarget.y < (camera->focalActorFloorHeight + 10.0f)) {
+        atTarget.y = camera->focalActorFloorHeight + 10.0f;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_Unique4.s")
+    Camera_ScaledStepToCeilVec3f(&atTarget, at, camera->atLerpStepScale, camera->atLerpStepScale, 0.2f);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_Unique5.s")
+    return 1;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_Unique0.s")
+s32 Camera_CalcAtForScreen(Camera* camera, VecSph* eyeAtDir, f32 yOffset, f32* focalActorPosY, f32 deltaYMax) {
```

Comment:
Went ahead and changed these back from `Camera_CalcLookAt` to `Camera_CalcAt` for consistency, but still open to suggestions for these set of names